src/main/java/edu/greenblitz/tobyDetermined/commands/telescopicArm/moveByNodes/GeneratePath.java

```diff
@@ -0,0 +1,32 @@
+package edu.greenblitz.tobyDetermined.commands.telescopicArm.moveByNodes;
+
+import edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.SystemsState;
+import edu.greenblitz.tobyDetermined.Nodesssss.NonCollidingNodeSystemAlgorithm.AStar;
+import edu.greenblitz.utils.GBCommand;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+
+import java.util.LinkedList;
+import java.util.function.Supplier;
+
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.CreateCurrents.system1CurrentNode;
+
+public class GeneratePath implements Supplier<Command> {
+
+    private final SystemsState targetNode;
+
+    public GeneratePath(SystemsState endNode) {
+        targetNode = endNode;
+    }
+
+    public Command get() {
+        SystemsState start = system1CurrentNode.getCurrentNode();
+        LinkedList<SystemsState> path = AStar.getPath(start, targetNode);
+        GBCommand[] nodeCommands = new GBCommand[path.size() - 1];
+        for (int i = 0; i < path.size() - 1; i++) {
+            nodeCommands[i] = new StraightLineMotion(path.get(i), path.get(i + 1));
+        }
```

Comment:
Not that bad but try finding a cleaner way to do this

Reply:
I couldn’t think about a way. Do you have a suggestion?

Reply:
Google "IntStream"

Reply:
Even using a foreach loop would be better

Reply:
instead of use IntStream i can do 
path.foreach( state -> ... )

butt i cant foreach here. i will need a prevoius variable and instead of array use list or create counter and thats wayyyyy less cleaner. unless you have an idea how can i know in which index of the array i need to put the new command withot counter?

Reply:
You're right, foreach won't work

I imagined something like this:

```java
GBCommand[] nodeCommands = IntStream.range(0, path.size() - 1)
                                    .mapToObj(i -> new StraightLineMotion(path.get(i), path.get(i + 1)))
                                    .toArray(GBCommand[]::new);
```